### PR TITLE
chore(bridge): Improve Bridge logging levels and messages

### PR DESCRIPTION
**Pull request by thisthat**
Signed-off-by: Giovanni Liva <giovanni.liva@dynatrace.com>

Part of #9111

### CODE REVIEW

**Review by heinzburgstaller — APPROVED** [review]

File: bridge/server/app.ts [hunk 1]
```diff
@@ -309,9 +312,25 @@ function handleError(err: any, req: Request, res: Response, authType: AuthType):
...
+    delete filteredHeaders[item];
+  }
+  // remove additional sec headers
+  Object.keys(headers).forEach((key) => {
+    if (key.startsWith('sec')) {
+      delete filteredHeaders[key];
+    }
+  });
+  return filteredHeaders;
+}
```
Comment (heinzburgstaller) [inline bridge/server/app.ts:334]: You can try to be more functional here. Something like:
```ts
const record = {...headers};
const filtered = Object.entries(record)
  .filter(entry => !(entry[0].startsWith('sec') || ['cookies', 'user-agent'].includes(entry[0])))
  .reduce((prev, current) => ({...prev, [current[0]]: current[1]}), {});
return filtered;
```

File: bridge/server/app.ts [hunk 2]
```diff
@@ -134,7 +137,7 @@ async function setBasicAUTH(app: Express, auth: AuthConfig): Promise<void> {
     userIP = userIP instanceof Array ? userIP[0] : userIP;
 
     if (userIP && isIPThrottled(userIP, auth)) {
-      log.error('Request limit reached');
+      log.error(`Request limit reached for IP ${userIP}`);
```
Comment (heinzburgstaller) [inline bridge/server/app.ts:140]: Is IP logging allowed? GDPR wise?

File: bridge/server/app.ts [hunk 3]
```diff
@@ -309,9 +312,25 @@ function handleError(err: any, req: Request, res: Response, authType: AuthType):
   }
 
   printError(err);
-  log.info(`Response status: ${err.response?.status}`);
+  log.error(`Response status ${err.response?.status} for ${req.method} ${req.url}`);
 
   return err.response?.status || 500;
 }
 
+function filterHeaders(headers: IncomingHttpHeaders): IncomingHttpHeaders {
```
Comment (heinzburgstaller) [inline bridge/server/app.ts:320]: A unit test would be nice

File: bridge/server/app.ts (hunk 1 above)
Comment (thisthat) [reply bridge/server/app.ts:334]: I think it makes very hard to follow the logic and purpose of the code.

File: bridge/server/app.ts (hunk 2 above)
Comment (thisthat) [reply bridge/server/app.ts:140]: I know there is an exception for data collection if it is for legitimate interest, which I guess this case fits since someone is trying to misuse the product. However, better be safe than sorry :) I will remove it to be on the safe side

File: bridge/server/app.ts (hunk 1 above)
Comment (heinzburgstaller) [reply bridge/server/app.ts:334]: Then just add named intermediate steps:
```ts
const record = {...headers};
const removeSecurityRelatedHeaders = (entry) =>  !(entry[0].startsWith('sec') || ['cookies', 'user-agent'].includes(entry[0]));
const toRecord = (prev, current) => ({...prev, [current[0]]: current[1]});
const filtered = Object.entries(record)
  .filter(removeSecurityRelatedHeaders)
  .reduce(toRecord, {});
return filtered;
```

Comment (thisthat) [reply bridge/server/app.ts]: I agree that this is more readable. However, the whole code base is built without a functional and I would propose to follow similar code pattern that are already there.

**Review by heinzburgstaller — APPROVED** [review]
